lib/mix/tasks/gh_pages_docs.ex

```diff
@@ -10,36 +10,82 @@ defmodule Mix.Tasks.GhPagesDocs do
   @shortdoc "Update a documentation version on GH Pages."
 
   @spec run(term) :: :ok
-  def run(_args) do
-    version = "v#{Project.config()[:version]}"
-
-    case File.read!("assets/js/versions.js") |> String.contains?(version) do
-      true ->
-        Mix.Task.run("docs")
-        File.cd!("doc")
-        Mix.shell().cmd("git stash")
-
-        Mix.shell().cmd("git checkout gh-pages")
-
-        case File.mkdir("../#{version}") do
-          result when result in [:ok, {:error, :eexist}] ->
-            Mix.shell().cmd("cp -r ./* ../#{version}")
-            File.cd!("..")
-            Mix.shell().cmd("ls")
-            Mix.shell().cmd("git add #{version}/*")
-            Mix.shell().cmd("git commit -m \"Update #{version}\"")
-            Mix.shell().cmd("rm -rf doc")
-            Mix.shell().cmd("git push origin gh-pages")
-
-          _ ->
-            Mix.raise("Cannot create the #{version} directory")
+  def run([version]) do
+    unless File.read!("assets/js/versions.js") |> String.contains?(version) do
+      update_versions_js()
+    end
+
+    version =
+      case version do
+        "latest" -> Project.config()[:version]
+        tag -> "v#{tag}"
+      end
+
+    Mix.Task.run("docs")
+    File.cd!("doc")
+    Mix.shell().cmd("git stash")
```

Comment:
At first, I thought that after `git stash` we loose the just generated docs. Then, I realised that the `doc` dir is added to `.gitignore` so `git stash` doesn't touch it. Hence, we can access it when we switch to the `gh-pages` branch. I think it'd clearer for other colleagues reading this script if a comment similar to this is added to the code.